invoice/templates/invoice/invoice_form.html

```diff
@@ -71,15 +71,15 @@ <h1 class="modal-title fs-5" id="InvoiceItemModalLabel">Add line item</h1>
                     <button type="button" class="btn-close" data-bs-dismiss="modal"
                             aria-label="Close"></button>
                 </div>
-                <form role="form" method="post" action="/invoice-item/add/">
+                <form role="form" method="post" action="/invoice/{{ invoice.id }}/add_item/">
                     {% csrf_token %}
                     <div class="modal-body">
-                        {% bootstrap_form invoice_item_form layout="floating" %}
-                    </div>
-                    <div class="modal-footer">
-                        {% bootstrap_button button_type="reset" content="Reset" %}
-                        {% bootstrap_button button_type="submit" content="Save" %}
-                    </div>
+                        {% bootstrap_form invoice_item_form layout="floating" exclude="invoice" %}
+                        <input type="hidden" name="invoice" value="{{ invoice.id }}">
```

Comment:
you could totally remove this form field set this in the view before saving. Because you don't need this field when you use the url which already contains that id.

Or you could pre-populate this field from python so you dont have to set it manually in html.

Reply:
I think this is less elegant as you wouldn't be able to use the Django magic.
Lot's of boilerplate must be added. Like saving the instance but not the form.
This is also complicates future implementations of user rights.